File: basics/structures/graph.java
import java.util.HashMap;

class Vertex{
  private String name;
  private int data;
  private HashMap<String, Integer> edges;

  public Vertex(String name){
    name = name;
    data = 0;
    edges = new HashMap<String, Integer>();
  }

  public String getName(){
    return name;
  }

  public int getData(){
    return data;
  }

  public HashMap<String, Integer> getEdges(){
    return edges;
  }

  public void setName(String newname){
    this.name = newname;
  }

  public void setData(int newdata){
    this.data = newdata;
  }

  public void addEdge(String connecting, int weight){
    edges.put(connecting, weight);
  }

  public int getWeight(String connected){
    if(edges.containsKey(connected)){
      return edges.get(connected);
    }
    return 0;
  }

}

class Graph{
  private HashMap<String, Vertex> vertices;
  private int size;

  public Graph(){
    vertices = new HashMap<String, Vertex>();
    size = 0;
  }

  public void addVertex(String name){
    Vertex newvertex = new Vertex(name);
    vertices.put(name, newvertex);
    size++;
  }

  public Vertex getVertex(String name){
    if(vertices.containsKey(name)){
      return vertices.get(name);
    }
    return null;
  }

  public void addEdge(String name1, String name2){
    if(vertices.containsKey(name1) && vertices.containsKey(name2)){
      Vertex v1 = vertices.get(name1);
      Vertex v2 = vertices.get(name2);
      v1.addEdge(v2.getName(), 0);
      v2.addEdge(v1.getName(), 0);
    }
  }

  public void addEdge(String name1, String name2, int weight){
    if(vertices.containsKey(name1) && vertices.containsKey(name2)){
      Vertex v1 = vertices.get(name1);
      Vertex v2 = vertices.get(name2);
      v1.addEdge(v2.getName(), weight);
      v2.addEdge(v1.getName(), weight);
    }
  }
}

File: basics/structures/graph.py
class Vertex:

    def __init__(self, name):
        self.name = name
        self.data = None
        self.edges = {}

    def getName(self):
        return self.name

    def getData(self):
        return self.data

    def getEdges(self):
        return edges

    def setName(self, newname):
        self.name = newname

    def setData(self, newdata):
        self.data = newdata

    #connecting = name of the node connecting to
    def addEdge(self, connecting, weight):
        self.edges[connecting] = weight

    def getWeight(self, connected):
        if connected in self.edges.keys():
            return self.edges[connected]
        return 0

    def __str__(self):
        return(str(self.name) + ":\t" + str(self.data))

class Graph:

    #dict<String, Vertex>
    def __init__(self):
        self.vertices = {}
        self.size = 0

    def addVertex(self, name):
        newvertex = Vertex(name)
        vertices[name] = newvertex
        size += 1

    def getVertex(self, name):
        if name in vertices.keys():
            return vertices[name]
        return None

    def getVertices(self):
        return self.vertices

    def addEdge(self, name1, name2, weight=0):
        if name1 in self.vertices.keys() and name2 in self.vertices.keys():
            v1 = self.vertices[name1]
            v2 = self.vertices[name2]
            v1.addEdge(v2.getName(), weight)
            v2.addEdge(v1.getName(), weight)
        return
